### DISCUSSION TITLE
Backup issue

### DISCUSSION THREAD

**Discussion by Stuntflyer** [mikeberger/borg_calendar#183] [General]
I have setup this option for backup in preferences.
![Borg](https://user-images.githubusercontent.com/46932988/230740764-2eb5dea2-db3d-4a35-b681-3b19f641f210.JPG)

If I choose to cancel this window during shutdown.
![Borg_2](https://user-images.githubusercontent.com/46932988/230740807-fd099331-8c85-4aed-9557-00e7c07aa073.JPG)

This is the error msg I get in "Robocopy".
![Borg_3](https://user-images.githubusercontent.com/46932988/230740857-2ed7fff9-70ee-44e3-aa48-0f9f05d16d02.JPG)

If I choose "okay" instead of cancel, robocopy does not error out.

Thanks, Mike

I think I solved it. Even if I don't want to back up I would still hit "Ok" , but without checking the box.

**Comment by mikeberger** [comment]
That's a bug. The cancel button should just get rid of the popup window, but not dispose of the main calendar window. 
In your case, the main window is being disposed of, but the program is still running with no open windows and still has the database locked.

The bug is only present under the following conditions:

- System Tray is not enabled
- Backup prompt is enabled
- Program close is started by hitting the window close (X) button and not via the Exit option in the menu.

I'm not sure when this bug crept into the code - probably when getting the close behavior to work properly when the system tray icon is enabled. I doubt I ever tested with the system tray icon disabled.

I'll create an issue from this discussion.

**Comment by mikeberger** [comment]
> I think I solved it. Even if I don't want to back up I would still hit "Ok" , but without checking the box.

The prompt window is very confusing. I think that I was being lazy and just trying to use the standard Java OK/Cancel dialog rather than doing it correctly. It's very unclear that OK and Cancel refer to the shutdown, while the checkbox adds an optional backup only if you select OK.

**Reply by Stuntflyer** [reply]
Mike, Thanks!
